### PR TITLE
Feature/no usage mode

### PR DESCRIPTION
**Pull request by petersteneteg**
Changes proposed in this PR:
 * Removes the old never functional Usage Mode
 * Adds a new dialog for configuring which properties that should be shown in a compositeprocessor
image

### CODE REVIEW

File: modules/base/src/algorithm/image/imagecontour.cpp [hunk 1]
```diff
@@ -28,12 +28,106 @@
...
+namespace {
+struct ImageContourDispatcher {
+    using type = std::shared_ptr<Mesh>;
+    template <typename Result, typename T>
+    std::shared_ptr<Mesh> operator()(const LayerRepresentation* in, size_t channel, double isoValue,
+                                     vec4 color);
+};
+
+template <typename Result, class DataType>
+std::shared_ptr<Mesh> ImageContourDispatcher::operator()(const LayerRepresentation* in,
```
Comment (ResearchDaniel) [inline modules/base/src/algorithm/image/imagecontour.cpp:47]: Is it not necessary to have the template code in the header?

File: include/inviwo/core/network/processornetwork.h [hunk 2]
```diff
@@ -186,6 +188,11 @@ class IVW_CORE_API ProcessorNetwork : public Serializable,
     template <typename C>
     void forEachProcessor(C callback);
 
+    auto processorView() const {
```
Comment (ResearchDaniel) [inline include/inviwo/core/network/processornetwork.h]: Any reason why we should not use:
processorRange
connectionRange
linkRange

File: modules/base/src/processors/imagecontourprocessor.cpp [hunk 3]
```diff
@@ -37,25 +39,24 @@ const ProcessorInfo ImageContourProcessor::processorInfo_{
     "Image Contour",                     // Display name
     "Image Processing",                  // Category
     CodeState::Experimental,             // Code state
-    Tags::None,                          // Tags
-};
+    Tags::CPU,                           // Tags
+    R"(Does marching squares on the image to extract a contour mesh.)"_unindentHelp};
```
Comment (ResearchDaniel) [inline modules/base/src/processors/imagecontourprocessor.cpp]:
```suggestion
    R"(Extracts a contour line for a specific value in the image using marching squares. The output contour is provided as a line mesh.)"_unindentHelp};
```

File: modules/base/src/processors/imagecontourprocessor.cpp [hunk 4]
```diff
@@ -37,25 +39,24 @@ const ProcessorInfo ImageContourProcessor::processorInfo_{
...
 ImageContourProcessor::ImageContourProcessor()
     : Processor()
-    , image_("image", true)
-    , mesh_("mesh")
-    , channel_("channel", "Channel", 0, 0, 4)
-    , isoValue_("iso", "ISO Value", 0.5, 0, 1)
-    , color_("color", "Color", vec4(1.0)) {
+    , image_("image", "Input image"_help, OutportDeterminesSize::Yes)
+    , mesh_("mesh", "Contour mesh"_help)
+    , channel_("channel", "Channel", "The image channel to use compare the iso value to"_help, 0,
```
Comment (ResearchDaniel) [inline modules/base/src/processors/imagecontourprocessor.cpp]:
```suggestion
    , channel_("channel", "Channel", "The image channel used to extract the contour"_help, 0,
```

File: modules/base/src/processors/imagecontourprocessor.cpp [hunk 5]
```diff
@@ -37,25 +39,24 @@ const ProcessorInfo ImageContourProcessor::processorInfo_{
...
+               {0, ConstraintBehavior::Immutable}, {4, ConstraintBehavior::Editable})
+    , isoValue_("iso", "ISO Value", "The contour iso value"_help, 0.5,
+                {0, ConstraintBehavior::Ignore}, {1, ConstraintBehavior::Ignore})
+    , color_("color", "Color",
+             util::ordinalColor(vec4(1.0)).set("The color of the resulting mesh"_help)) {
```
Comment (ResearchDaniel) [inline modules/base/src/processors/imagecontourprocessor.cpp]:
```suggestion
             util::ordinalColor(vec4(1.0)).set("The contour color"_help)) {
```

File: src/core/processors/compositeprocessorutils.cpp [hunk 6]
```diff
@@ -283,7 +285,7 @@ void util::expandCompositeProcessorIntoNetwork(CompositeProcessor& composite) {
             network.addLink(l);
         }
 
-        network.removeProcessor(&composite);
+        auto tmp = network.removeProcessor(&composite);
```
Comment (ResearchDaniel) [inline src/core/processors/compositeprocessorutils.cpp]: Does tmp have something to do with deleting the composite?
Either name it accordingly or add a comment in that case.

File: src/qt/editor/subpropertyselectiondialog.cpp [hunk 7]
```diff
@@ -0,0 +1,687 @@
...
+            return Qt::ItemIsEnabled;
+        }
+    }
+
+    virtual QVariant data(const QModelIndex& index, int role) const override {
+        if (role ==  Qt::ForegroundRole) {
+            return {};
+        }
+    
+        if (index.column() == 3) {
```
Comment (ResearchDaniel) [inline src/qt/editor/subpropertyselectiondialog.cpp]: These magic numbers could perhaps be named?

File: src/qt/editor/subpropertyselectiondialog.cpp [hunk 8]
```diff
@@ -0,0 +1,687 @@
...
+
+        QSizePolicy sp = tree->sizePolicy();
+        sp.setHorizontalStretch(3);
+        sp.setVerticalStretch(3);
+        sp.setVerticalPolicy(QSizePolicy::Expanding);
+        sp.setHorizontalPolicy(QSizePolicy::MinimumExpanding);
+        tree->setSizePolicy(sp);
+    };
+
+    layout->addWidget(new QLabel("Composite Network"), 0, 0);
```
Comment (ResearchDaniel) [inline src/qt/editor/subpropertyselectiondialog.cpp]:
```suggestion
    layout->addWidget(new QLabel("All Properties"), 0, 0);
```

**Review by ResearchDaniel — CHANGES_REQUESTED** [review]
Very nice! Some minor comments.
Could you also update the changelog and state that we now advice people to use CanvasWithProperties instead of the UsageMode

File: modules/base/src/algorithm/image/imagecontour.cpp (hunk 1 above)
Comment (petersteneteg) [reply modules/base/src/algorithm/image/imagecontour.cpp:47]: Only if you plan on using it in some other translation unit. this is only used further down in this translation unit, then there is no need to have it in the header....

File: include/inviwo/core/network/processornetwork.h (hunk 2 above)
Comment (petersteneteg) [reply include/inviwo/core/network/processornetwork.h]: no particular reason, can change

File: src/core/processors/compositeprocessorutils.cpp (hunk 6 above)
Comment (petersteneteg) [reply src/core/processors/compositeprocessorutils.cpp]: don't think is was needed with the holder up top

**Review by ResearchDaniel — DISMISSED** [review]

**Review by ResearchDaniel — APPROVED** [review]

**Pull request merged**